src/Razor/test/Microsoft.VisualStudio.Razor.IntegrationTests/ProjectTests.cs

```diff
@@ -155,4 +155,78 @@ public async Task OpenExistingProject()
         var actualProjectFileName = await TestServices.SolutionExplorer.GetAbsolutePathForProjectRelativeFilePathAsync(RazorProjectConstants.BlazorProjectName, RazorProjectConstants.ProjectFile, ControlledHangMitigatingCancellationToken);
         Assert.Equal(expectedProjectFileName, actualProjectFileName);
     }
+
+    [IdeFact(MaxAttempts = 1)]
+    public async Task OpenExistingProject_WithReopenedFile()
```

Comment:
I've added this test, and the one below, just in case, but it turns out they don't fail even without the fixes in this PR, because of other issues. Have started a Teams thread about it.